src/passes/RemoveUnusedModuleElements.cpp

```diff
@@ -333,24 +333,27 @@ struct RemoveUnusedModuleElements : public Pass {
     //
     // We can only do this in a closed world, as otherwise function references
     // may be called outside of the module (if they escape, which we could in
-    // principle track, see the TODO earlier in this file).
+    // principle track, see the TODO earlier in this file). So in the case of an
+    // open world we should not have noted anything in uncalledRefFuncMap
+    // earlier and not do any related optimizations there.
+    if (!closedWorld) {
+      assert(analyzer.uncalledRefFuncMap.empty());
+    }
```

Comment:
```suggestion
    assert(closedWorld || analyzer.uncalledRefFuncMap.empty());
```
Super minor nit... Feel free to ignore if you see this after CI finishes.